shards/shards.go

```diff
@@ -971,6 +971,14 @@ func mkRankedShard(s zoekt.Searcher) *rankedShard {
 			if priority > maxPriority {
 				maxPriority = priority
 			}
+
+			// No one is reading s yet, so we can mutate it.
+			// zoekt-sourcegraph-indexserver does not set Rank so at this point we
+			// treat Priority like star count and set it.
+			if repo.Rank == 0 && priority > 0 {
+				l := math.Log(float64(priority))
+				repo.Rank = uint16((1.0 - 1.0/math.Pow(1+l, 0.6)) * 10000)
```

Comment:
Can this magic be documented somewhat?

Reply:
![image](https://user-images.githubusercontent.com/103087/199010269-49549073-29b9-4e37-b06f-d1fba17a5009.png)

...but y tho

Reply:
This is copy pasta straight from the mind of Han-Wen. Not allowed to question it :p

No idea why. But given how we mix the value in, I just wanna retain backwards compatability incase experimentation weights this value too high or low. IE lets follow up :)

Reply:
where... did you get it though?

Reply:
"upstream": https://github.com/google/zoekt/blob/master/gitindex/index.go#L247-L250

Reply:
> where... did you get it though?

sorry you might of missed due to race conditions and timestamps I see here: https://github.com/google/zoekt/blob/master/gitindex/index.go#L247-L250